src/utils/image.ts

```diff
@@ -67,3 +74,50 @@ export const getFirebaseStorageResizedImage = (
     url,
     options
   );
+
+// @see https://crypto.stackexchange.com/questions/8533/why-are-bitwise-rotations-used-in-cryptography/8534#8534
+const DIFFUSION_PRIME = 31;
+
+type DetermineAvatar = (options?: {
+  avatars?: string[];
+  email?: string;
+  index?: number;
+  partyName?: string;
+  pictureUrl?: string;
+  userInfo?: firebase.UserInfo;
+  user?: User;
+}) => string;
+
+export const determineAvatar: DetermineAvatar = (options) => {
+  const { avatars, email, index, partyName, pictureUrl, user, userInfo } =
+    options ?? {};
+  const list = avatars ?? DEFAULT_AVATAR_LIST;
+  const url = pictureUrl || user?.pictureUrl || "";
+
+  // checking user.anonMode ruled against by https://github.com/sparkletown/internal-sparkle-issues/issues/1615#issuecomment-991054012
+
+  if (url.startsWith("/static") || url.startsWith("http")) {
+    return url;
+  }
+
+  if (isDefined(index) && Number.isSafeInteger(index) && index >= 0) {
+    return list[index % list.length];
+  }
```

Comment:
Given that this is only used in one place and this place randoms the index, can we remove this just to use the seed & hash implementation for a cleaner view?

Reply:
I went with the simplest solution first so that we get more consistent output and there are number of fallbacks that go from the most to the least consistent. so far, the index is my least favorite ones, but had to add because of AnimateMap.

I say this PR is good enough for solving the current problem and the hashing can be improved once all the places that require an avatar have similar data to provide.

Reply:
The above rationalization is more or less the same with the image checks. If and when we need it, can be added with a subsequent PR. Right now it might just be an overkill and OOS

Reply:
Ideally, what I envisioned for a future PR was the addition of another field in DB that sets default avatar once, on user creation, and picks that one consistently if/when the chosen avatar fails.